homeassistant/components/netatmo/camera.py

```diff
@@ -334,82 +305,91 @@ def _enable_motion_detection(self, enable):
                 self._data.update()
-                (self._vpnurl, self._localurl) = self._data.camera_data.cameraUrls(
-                    camera=self._camera_name
+                (self._vpnurl, self._localurl) = self._data.camera_data.camera_urls(
+                    cid=self._camera_id
                 )
                 return None
         except requests.exceptions.RequestException as error:
             _LOGGER.error("Welcome/Presence URL changed: %s", error)
             self._data.update()
-            (self._vpnurl, self._localurl) = self._data.camera_data.cameraUrls(
-                camera=self._camera_name
+            (self._vpnurl, self._localurl) = self._data.camera_data.camera_urls(
+                cid=self._camera_id
             )
             return None
         else:
             self.async_schedule_update_ha_state(True)
 
-    # Netatmo Presence specific camera method.
 
-    def set_light_auto(self):
-        """Set flood light in automatic mode."""
-        _LOGGER.debug(
-            "Set the flood light in automatic mode for the camera '%s'", self._name
-        )
-        self._set_light_mode("auto")
+class CameraData:
+    """Get the latest data from Netatmo."""
+
+    def __init__(self, hass, auth):
+        """Initialize the data object."""
+        self._hass = hass
+        self.auth = auth
+        self.camera_data = None
+        self.camera_ids = []
+        self.module_names = []
+        self.camera_type = None
+
+    def get_camera_home_id(self, camera_id):
+        """Return the home id for a given camera id."""
+        for home_id in self.camera_data.cameras:
+            for camera in self.camera_data.cameras[home_id].values():
+                if camera["id"] == camera_id:
+                    return home_id
+        return None
 
-    def set_light_on(self):
-        """Set flood light on."""
-        _LOGGER.debug("Set the flood light on for the camera '%s'", self._name)
-        self._set_light_mode("on")
+    def get_all_camera_ids(self):
+        """Return all camera available on the API as a list."""
+        self.camera_ids = []
+        self.update()
+        for home_id in self.camera_data.cameras:
+            for camera in self.camera_data.cameras[home_id].values():
+                self.camera_ids.append(camera["id"])
+        return self.camera_ids
+
+    def get_module_names(self, camera_id):
+        """Return all module available on the API as a list."""
+        self.module_names = []
+        self.update()
+        for module in self.camera_data.modules.values():
+            if camera_id == module["cam_id"]:
+                self.module_names.append(module["name"])
+        return self.module_names
+
+    def get_camera_type(self, camera_id=None):
+        """Return camera type for a camera, cid has preference over camera."""
+        self.camera_type = self.camera_data.cameraType(cid=camera_id)
+        return self.camera_type
+
+    def get_persons(self):
```

Comment:
This method name doesn't return anything.

Reply:
Old code. I renamed the method.